src/contracts/core/AVSDirectory.sol

```diff
@@ -276,9 +276,18 @@ contract AVSDirectory is
         uint32 effectTimestamp = uint32(block.timestamp) + ALLOCATION_DELAY;
         for (uint256 i = 0; i < allocations.length; ++i) {
             // 1. For the given (operator,strategy) clear all pending free magnitude for the strategy and update freeMagnitude
-            _updateFreeMagnitude(operator, allocations[i].strategy);
+            // numToComplete = 0 defaults to completing all pending deallocations up to uint8 max (256)
+            _updateFreeMagnitude({
```

Comment:
shouldn't we accept `numToComplete` to this interface?

Reply:
leaving out of interface for simplicity, if gas costs are an issue then can call `updateFreeMagnitude` separately